.github/scripts/tag_check_readme_update.py

```diff
@@ -0,0 +1,385 @@
+    #so we need to use the Github data API:
+    changelog_git_blob = cam_repo.get_git_blob(changelog_sha)
+
+    #Convert changelog blob to unicode string:
+    changelog = base64.b64decode(changelog_git_blob.content).decode("UTF-8")
+
+    #+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+    #Search for tag in ChangeLog, to ensure log was properly updated
+    #+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+    #Make ChangeLog all lowercase, so that searching is case independent:
+    changelog = changelog.lower()
+
+    #Find first occurence of "Tag name" string:
+    tag_log_idx = changelog.find("tag name")
+
+    if tag_log_idx == -1:
+        endmsg = "No 'tag name' entry found in the ChangeLog!\n"
+        endmsg += "The ChangeLog file has likely been corrupted."
+        end_script_fail(cam_pword, endmsg)
+    else:
+        #Find first occurence of "Originator", to set tag search end:
+        orig_log_idx = changelog.find("originator")
+        if orig_log_idx == -1:
+            endmsg = "No 'Originator' entry found in the ChangeLog!\n"
+            endmsg += "The ChangeLog file has likely been corrupted."
+            end_script_fail(cam_pword, endmsg)
+
+    #Search for tag string inbetween "tag name" and "originator":
+    tag_found = changelog[tag_log_idx:orig_log_idx].find(tag_name)
+
+    if tag_found == -1:
+        endmsg = "Newly-created tag '{}' not the first tag listed at top of ChangeLog!\n"
+        endmsg += "Please update the ChangeLog so that the newest tag is listed first."
+        end_script_fail(cam_pword, endmsg.format(tag_name))
+
+    #+++++++++++++++++++++++++++++++++
+    #Extract README file contents/text
+    #+++++++++++++++++++++++++++++++++
+
+    #Grab README file object:
+    #Note: This command always uses the default branch unless
+    #      specifically told to do otherwise.
+    readme_obj = cam_repo.get_contents("README.md")
+
+    #Extract README content (as a bytestring):
+    readme_content = readme_obj.decoded_content
+
+    #Convert bytestring to unicode (regular) string:
+    readme_text = readme_content.decode('UTF-8')
+
+    #+++++++++++++++++++++++++++++++++
+    #Upate README file development tag
+    #+++++++++++++++++++++++++++++++++
+
+    #Compile development tag text expression:
+    dev_tag_pattern = re.compile(r'cam\d+_\d+_\d+')
+
+    #Search for tag text in README file:
+    dev_tag_match = dev_tag_pattern.search(readme_text)
+
+    #End script if nothing is found:
+    if not dev_tag_match:
+        endmsg = "No text matches expected development tag pattern in README.md file, "
+        endmsg += "so no further action will be taken."
+        end_script(endmsg)
+
+    #Extract start and end indices of matched tag text:
+    dev_tag_idx = dev_tag_match.span()
+
+    #Add new tag to README text:
+    new_content = readme_text[:dev_tag_idx[0]] + tag_name + readme_text[dev_tag_idx[1]:]
```

Comment:
Why not just use `replace`?
```suggestion
    #Add new tag to README text:
    new_content = readme_text.replace dev_tag_match.group(0), tag_name)
```